### PR TITLE
Use spark.write.csv in to_csv of Series and DataFrame

### PR DESCRIPTION
**Pull request by HyukjinKwon**
This PR proposes to use `spark.write.csv` API to enable distributed computation when `path` is specified. If `path` is not specified, it just calls pandas' `to_csv` as was.

Closes #677

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
Let me fix up the tests tomorrow in kst

File: databricks/koalas/tests/test_dataframe_conversion.py [hunk 1]
```diff
@@ -43,32 +43,6 @@ def strip_all_whitespace(str):
         """A helper function to remove all whitespace from a string."""
         return str.translate({ord(c): None for c in string.whitespace})
 
-    def test_csv(self):
```
Comment (HyukjinKwon) [inline databricks/koalas/tests/test_dataframe_conversion.py]: Moved to `test_csv.py`

File: databricks/koalas/generic.py [hunk 2]
```diff
@@ -456,118 +456,143 @@ def to_numpy(self):
...
-    def to_csv(self, path_or_buf=None, sep=",", na_rep='', float_format=None,
-               columns=None, header=True, index=True, index_label=None,
-               mode='w', encoding=None, compression='infer', quoting=None,
-               quotechar='"', line_terminator="\n", chunksize=None,
-               tupleize_cols=None, date_format=None, doublequote=True,
-               escapechar=None, decimal='.'):
-        """
+    def to_csv(self, path=None, sep=',', na_rep='', columns=None, header=True,
+               quotechar='"', date_format=None, escapechar=None, num_files=None,
+               **kwargs):
```
Comment (HyukjinKwon) [inline databricks/koalas/generic.py]: Hmm .. should I maybe not add this `kwargs` to allow PySpark' option directly for now @ueshin?

Comment (ueshin) [reply databricks/koalas/generic.py]: I think it's okay to have it. Actually I added `options` for `read_sql()/_table()/_query()`.

**Comment by HyukjinKwon** [comment]
Okie .. build passed.

**Comment by HyukjinKwon** [comment]
Okie .. build passed.

File: databricks/koalas/generic.py [hunk 3]
```diff
@@ -456,118 +458,142 @@ def to_numpy(self):
...
-        ...                    columns=['name', 'mask', 'weapon'])
-        >>> df.to_csv(index=False)
-        'name,mask,weapon\\nRaphael,red,sai\\nDonatello,purple,bo staff\\n'
-        >>> df.name.to_csv()  # doctest: +ELLIPSIS
-        '...Raphael\\n1,Donatello\\n'
+        >>> df = ks.DataFrame(dict(
+        ...    date=list(pd.date_range('2012-1-1 12:00:00', periods=3, freq='M')),
+        ...    country=['KR', 'US', 'JP'],
+        ...    code=[1, 2 ,3]), columns=['date', 'country', 'code'])
+        >>> df.sort_values(by="date")  # doctest: +ELLIPSIS, +NORMALIZE_WHITESPACE
```
Comment (ueshin) [inline databricks/koalas/generic.py]: `+ELLIPSIS` is not needed?

File: databricks/koalas/generic.py [hunk 4]
```diff
@@ -456,118 +458,142 @@ def to_numpy(self):
...
+        >>> print(df.to_csv())  # doctest: +NORMALIZE_WHITESPACE
+        date,country,code
+        2012-01-31 12:00:00,KR,1
+        2012-02-29 12:00:00,US,2
+        2012-03-31 12:00:00,JP,3
+
+        >>> df.to_csv(path=r'%s/to_csv/foo.csv' % path, num_files=1)
+        >>> ks.read_csv(
+        ...    path=r'%s/to_csv/foo.csv' % path
+        ... ).sort_values(by="date")  # doctest: +ELLIPSIS, +NORMALIZE_WHITESPACE
```
Comment (ueshin) [inline databricks/koalas/generic.py]: ditto.

File: databricks/koalas/generic.py [hunk 5]
```diff
@@ -456,118 +458,142 @@ def to_numpy(self):
...
+        >>> print(df.date.to_csv())  # doctest: +NORMALIZE_WHITESPACE
+        date
+        2012-01-31 12:00:00
+        2012-02-29 12:00:00
+        2012-03-31 12:00:00
+
+        >>> df.date.to_csv(path=r'%s/to_csv/foo.csv' % path, num_files=1)
+        >>> ks.read_csv(
+        ...     path=r'%s/to_csv/foo.csv' % path
+        ... ).sort_values(by="date")  # doctest: +ELLIPSIS, +NORMALIZE_WHITESPACE
```
Comment (ueshin) [inline databricks/koalas/generic.py]: ditto.

File: databricks/koalas/generic.py (hunk 3 above)
Comment (HyukjinKwon) [reply databricks/koalas/generic.py]: oh that's needed for `...` for index.

Comment (HyukjinKwon) [reply databricks/koalas/generic.py]: here:

```
Expected:
                       date
    ... 2012-01-31 12:00:00
    ... 2012-02-29 12:00:00
    ... 2012-03-31 12:00:00
Got:
                     date
    0 2012-01-31 12:00:00
    1 2012-02-29 12:00:00
    2 2012-03-31 12:00:00
```

**Comment by HyukjinKwon** [comment]
Thanks, @ueshin. Merged to master.

**Pull request merged**